### PR TITLE
Fixes and enhancements for array query param handling

### PR DESCRIPTION
**Pull request by chris48s**
This PR is a bit of a mixed bag of related things.

1. There is an infinite render loop on pages that use query params, causing a `Maximum Update Depth Exceeded` error.

2. Respect the `maxItems` property for array query params.

3. OpenAPI supports number of different [styles for array query params](https://swagger.io/docs/specification/serialization/#query) via the `style` and `explode` properties. This PR adds compatibility for those.

### CODE REVIEW

File: packages/docusaurus-theme-openapi/src/theme/ApiDemoPanel/ParamOptions/index.tsx [hunk 1]
```diff
@@ -184,7 +190,7 @@ function ParamArrayFormItem({ param }: ParamProps) {
         value: values.length > 0 ? values : undefined,
       })
     );
-  }, [dispatch, items, param]);
+  }, [dispatch, items]);
```
Comment (chris48s) [inline packages/docusaurus-theme-openapi/src/theme/ApiDemoPanel/ParamOptions/index.tsx:193]: The way this was written before there is an infinite render loop.
Removing `param` from the dependency array fixed the infinite loop, but doing this introduces an eslint error. I'm not a react expert, so not really sure where to go next with this one. I'd be really open to feedback on how to handle this better.

File: packages/docusaurus-theme-openapi/src/theme/ApiDemoPanel/ParamOptions/index.tsx [hunk 2]
```diff
@@ -230,9 +236,12 @@ function ParamArrayFormItem({ param }: ParamProps) {
...
       ))}
-      <button className={styles.buttonThin} onClick={handleAddItem}>
-        Add item
-      </button>
+      {(param?.schema?.maxItems == null ||
+        items.length < param.schema.maxItems) && (
+        <button className={styles.buttonThin} onClick={handleAddItem}>
+          Add item
+        </button>
+      )}
```
Comment (chris48s) [inline packages/docusaurus-theme-openapi/src/theme/ApiDemoPanel/ParamOptions/index.tsx:244]: I decided to just hide the "Add item" button once we reach the `maxItems` limit here, but another approach would be to show a diabled button instead, like this:

```jsx
{(param?.schema?.maxItems == null || items.length < param.schema.maxItems) &&
  <button className={styles.buttonThin} onClick={handleAddItem}>
    Add item
  </button>
  ||
  <button className={styles.buttonThin} disabled>
    Add item
  </button>
}
```

I don't have a really strong view on it.

File: packages/docusaurus-theme-openapi/src/theme/ApiDemoPanel/ParamOptions/index.tsx (hunk 1 above)
Comment (chris48s) [reply packages/docusaurus-theme-openapi/src/theme/ApiDemoPanel/ParamOptions/index.tsx]: Superseded by d616c10de8ff23c9750c9ad477431c775faf7da4

**Comment by chris48s** [comment]
I've circled back to this and pushed another commit which takes a different approach to fixing the infinite render loop. This gets rid of the loop without introducing any ESLint errors.

Would it be possible to get a review on this?

File: packages/docusaurus-theme-openapi/src/theme/ApiDemoPanel/ParamOptions/index.tsx (hunk 2 above)
Comment (bourdakos1) [reply packages/docusaurus-theme-openapi/src/theme/ApiDemoPanel/ParamOptions/index.tsx:251]: Could we do disabled instead? I think that’s more accessibility friendly

File: packages/docusaurus-theme-openapi/src/theme/ApiDemoPanel/ParamOptions/index.tsx [hunk 3]
```diff
@@ -161,10 +166,16 @@ function ArrayItem({
 }
 
 function ParamArrayFormItem({ param }: ParamProps) {
-  const [items, setItems] = useState<{ id: string; value?: string }[]>([]);
+  const [items, setItems] = useState<Item[]>([]);
   const dispatch = useTypedDispatch();
 
   function handleAddItem() {
```
Comment (bourdakos1) [inline packages/docusaurus-theme-openapi/src/theme/ApiDemoPanel/ParamOptions/index.tsx:172]: Do we need to call updateItem here?

File: packages/docusaurus-theme-openapi/src/theme/ApiDemoPanel/ParamOptions/index.tsx (hunk 2 above)
Comment (chris48s) [reply packages/docusaurus-theme-openapi/src/theme/ApiDemoPanel/ParamOptions/index.tsx]: :+1: done in 71ac129

File: packages/docusaurus-theme-openapi/src/theme/ApiDemoPanel/ParamOptions/index.tsx (hunk 3 above)
Comment (chris48s) [reply packages/docusaurus-theme-openapi/src/theme/ApiDemoPanel/ParamOptions/index.tsx:172]: Functionally I think it doesn't matter if we don't call it because at the point we add a new item it has no value yet, so we'd just pass over the value/strip it when we render the code examples. We only add it to the code example once it has a value (which is covered by the `handleChangeItem()` event).

An the other hand, the value of `items` has changed and calling `updateItems()` doesn't do any harm. So maybe from an "ideological" perspective, we should?

Again, I don't have a strong view on this one.

File: packages/docusaurus-theme-openapi/src/theme/ApiDemoPanel/ParamOptions/index.tsx [hunk 4]
```diff
@@ -230,9 +243,16 @@ function ParamArrayFormItem({ param }: ParamProps) {
...
+      {((param?.schema?.maxItems == null ||
+        items.length < param.schema.maxItems) && (
+        <button className={styles.buttonThin} onClick={handleAddItem}>
+          Add item
+        </button>
+      )) || (
+        <button className={styles.buttonThin} disabled>
+          Add item
+        </button>
+      )}
```
Comment (bourdakos1) [inline packages/docusaurus-theme-openapi/src/theme/ApiDemoPanel/ParamOptions/index.tsx:255]: This is a bit hard to follow. Can we do something like:

```tsx
<button 
  className={styles.buttonThin}
  onClick={handleAddItem}
  disabled={param?.schema?.maxItems != null && param.schema.maxItems <= items.length}
>
  Add item
</button>
```

**Review by bourdakos1 — APPROVED** [review]
